src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,39 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List getAll(Class type) {
+        if (type == Bulldozer.class) {
+            return new BulldozerProducer().get();
+        }
+        if (type == Excavator.class) {
+            return new ExcavatorProducer().get();
+        }
+        if (type == Truck.class) {
+            return new TruckProducer().get();
+        }
+        return new ArrayList<>();
+    }
```

Comment:
```suggestion
public class MachineServiceImpl implements MachineService<Machine> {
    private BulldozerProducer bulldozerProducer = new BulldozerProducer();
    private ExcavatorProducer excavatorProducer = new ExcavatorProducer();
    private TruckProducer truckProducer = new TruckProducer();

    @Override
    public List<Machine> getAll(Class type) {
        if (type == Bulldozer.class) {
            return new ArrayList<>(bulldozerProducer.get());
        }
        if (type == Excavator.class) {
            return new ArrayList<>(excavatorProducer.get());
        }
        if (type == Truck.class) {
            return new ArrayList<>(truckProducer.get());
        }
        return new ArrayList<>();
    }
```

отак буде працювати 🙂